### ISSUE TITLE
Does Typhoeus support HTTP/2

### ISSUE THREAD

**Issue by cschull** [typhoeus/typhoeus#690]
Hi,

I have some large assets I need to download (>2GB) with Typhoeus, but the downloads consistently fail using HTTP/1.1 and I've only gotten them to work with HTTP/2. Does Typhoeus support HTTP/2 and if so, how would I implement that?

**Comment by mmizutani** [comment]
Yes. Pass a `http_version: :httpv2_0` option to `Typhoeus::Request.new`.
https://github.com/typhoeus/ethon/blob/453c6f0ba37a7d42978c90f2399f5c2cd66b32a6/lib/ethon/curls/options.rb#L355

**Comment by semaperepelitsa** [comment]
It should use HTTP/2 by default where available, [same as curl](https://everything.curl.dev/http/http2).

```
> Typhoeus::Request.new("https://www.google.com", verbose: true).run

  Trying 142.250.178.4:443...
Connected to www.google.com (142.250.178.4) port 443 (#1)
...
Using HTTP2, server supports multiplexing
```

**Comment by jeremyhaile** [comment]
How do you multiplex requests? Does Hydra automatically multiplex requests when using an HTTP/2 server?
